rpc/provider/smoldot.ts

```diff
@@ -27,12 +43,21 @@ export const smoldotProvider: Provider<
     nextId,
     send: (message) => {
       (async () => {
-        const conn = await connection(chainSpec, listener);
+        let conn: SmoldotProviderConnection;
+        try {
+          conn = await connection(chainSpec, listener);
+        } catch (error) {
+          listener(new ProviderHandlerError(error as SmoldotHandlerErrorData));
+          return;
+        }
         try {
           return conn.inner.sendJsonRpc(JSON.stringify(message));
-        } catch (e) {
-          // TODO: funnel this to relevant handler(s)
-          return new ProviderSendError(e);
+        } catch (error) {
+          if (error instanceof MalformedJsonRpcError || error instanceof QueueFullError) {
+            listener(new ProviderSendError(error, message));
+            return;
```

Comment:
Why return here? If you get a not-all-branches-return err, a more fitting place for the `return` may be after line 61.